aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -687,6 +691,197 @@ public void onError(Exception error) {
+    ) {
+
+        awsMobileClient.updateUserAttributes(
+                Collections.singletonMap(attribute.getKey().getKeyString(), attribute.getValue()),
+                new Callback<List<UserCodeDeliveryDetails>>() {
+                    @Override
+                    public void onResult(List<UserCodeDeliveryDetails> result) {
+                        if (result.size() == 0) {
+                            onSuccess.accept(new AuthUpdateAttributeResult(
+                                    true,
+                                    new AuthNextUpdateAttributeStep(
+                                            AuthUpdateAttributeStep.DONE,
+                                            Collections.emptyMap(),
+                                            null)
+                            ));
+                        } else {
+                            for (UserCodeDeliveryDetails details : result) {
+                                onSuccess.accept(new AuthUpdateAttributeResult(
+                                        true,
+                                        new AuthNextUpdateAttributeStep(
+                                                AuthUpdateAttributeStep.CONFIRM_ATTRIBUTE_WITH_CODE,
+                                                Collections.emptyMap(),
+                                                convertCodeDeliveryDetails(details))
+                                ));
+                            }
+                        }
+                    }
+
+                    @Override
+                    public void onError(Exception error) {
+                        onError.accept(new AuthException(
+                                "Failed to update user attributes",
+                                error,
+                                "See attached exception for more details"
+                        ));
+                    }
+                }
+        );
+    }
+
+    @Override
+    public void updateUserAttributes(
+            @NonNull List<AuthUserAttribute> attributes,
+            @NonNull Consumer<Map<AuthUserAttributeKey, AuthUpdateAttributeResult>> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+        Map<String, String> attributesMap = new HashMap<>();
+        for (AuthUserAttribute attribute : attributes) {
+            attributesMap.put(attribute.getKey().getKeyString(), attribute.getValue());
+        }
+
+        awsMobileClient.updateUserAttributes(
+                attributesMap,
+                new Callback<List<UserCodeDeliveryDetails>>() {
+                    @Override
+                    public void onResult(List<UserCodeDeliveryDetails> result) {
+                        Map<AuthUserAttributeKey, AuthUpdateAttributeResult> resultMap =
+                                new HashMap<>();
+
+                        for (UserCodeDeliveryDetails details : result) {
+                            Map<String, UserCodeDeliveryDetails> codeDetailsMap =
```

Comment:
Defining a singleton map here defeats the purpose of having a map. My suggested logic was to do a loop through the `UserCodeDeliveryDetails` in `result` and add them all to a map before looping through the `attributesMap` to avoid having a nested loop like this one.